src/protocol/communication.rs

```diff
@@ -42,10 +42,10 @@ pub fn send(stream: &mut TcpStream, message: &serde_json::Value) -> BoxResult<()
     let serialised_message = serde_json::to_vec(message)?;
 
     log::debug!(
-        "sending message of length {}, {:?}, to {}...",
+        "sending message to {} length {}, {:?}...",
```

Comment:
There should be a comma or other delimiter here to break up the elements: "sending message to {}**,** length..."